simtools/util/general.py

```diff
@@ -770,3 +775,72 @@ def get_log_excerpt(log_file, n_last_lines=30):
         f"{collect_final_lines(log_file, n_last_lines)}\n\n"
         "====================================================================\n"
     )
+
+
+def convert_2D_to_radial_distr(xaxis, yaxis, hist2d, bin_size=50, max_dist=1000):
+    """
+    Convert a 2D histogram of the photon positions on the ground to a 1D radial distribution.
+
+    Parameters
+    ----------
+    xaxis: numpy.array
+        The values of the X axis (histogram edges) on the ground.
+    yaxis: numpy.array
+        The values of the Y axis (histogram edges) on the ground.
+    hist2d: numpy.ndarray
+        The histogram
+    bin_size: float
+        Size of the step in distance (in meters).
+    max_dist: float
+       Maximum distance to consider in the 1D histogram (in meters).
+
+    Returns
+    -------
+    np.array
+        The edges of the 1D histogram in meters with size = int(max_dist/bin_size) + 1.
+    np.array
+        The values of the 1D histogram with size = int(max_dist/bin_size).
+    """
+
+    distance2D = np.empty_like(hist2d)
```

Comment:
Could I ask you to please add some comments to this function to explain the steps of the algorithm? It's not easy to follow without actually running through some examples.

Reply:
I added some comments. I hope that helps?